### DISCUSSION TITLE
Custom ShareX Uploader

### DISCUSSION THREAD

**Discussion by KloudAlpha** [forscht/ddrive#54] [Q&A]
I'm trying to create a custom ShareX uploader for my ddrive instance, mostly so I can queue uploads. So far is what I've got, but I've hit a wall. I feel like I'm missing something at URL parameters but I don't know what.

![2022-12-29-21 45 00-ShareX](https://user-images.githubusercontent.com/11934011/210029333-0322e57e-188e-4742-a2a3-572f232084af.png)

 When I test the upload with this configuration I get this error: (409) Conflict

![2022-12-29-20 39 05-ShareX](https://user-images.githubusercontent.com/11934011/210026182-b3397905-8f2f-4cb7-895d-c8c776ed3448.png)

Any ideas on how to get this working? My request url is the same as I'd use with curl -X POST.

Also, if anyone else is as nooby as me and uses authorization for their ddrive instance, to supply the username and password you make a header entry with the name "Authorization" and the value is "Basic" then your username:password encoded as Base64. i.e. if your username and password were username:password then the value would be "Basic dXNlcm5hbWU6cGFzc3dvcmQ=" You can use the ShareX Hash check tool to encode your credentials securely.

**Comment by forscht** [comment]
I think you're passing invalid URL, are you sure variable `{filename}` is getting populated correctly?

**Reply by KloudAlpha** [reply]
I was able to get uploads working by appending {filename} to the end of the request url. However I could not get authorization to work.. Auth worked fine for actually transferring the files, but for whatever reason the http response was always 401 Unauthorized, despite the fact that it successfully uploaded.

**Reply by KloudAlpha** [reply]
https://gist.github.com/KloudAlpha/27825abf9c3394c9e2f0fb8a2637688e
The configuration I was using with the authorization error.
